### PR TITLE
chore(uve): Implement Language selector on new FTM toolbar

### LINKED ISSUE #30735: FTM: New UVE Toolbar - Implement Language selector
### Parent Issue

https://github.com/dotCMS/core/issues/30598

### Task

Adapt the existing Language Selector component for the new UVE Toolbar, maintaining its current functionality and interactions. This task involves adapting the `newWithEditorToolbar()` 

### Proposed Objective

Successfully integrate the Language Selector into the new toolba while preserving all existing functionality across both traditional and headless pages.

### Acceptance Criteria

- Ensure all existing Language Selector functionality remains intact:
  - Language switching
  - Language list display
  - Current language indication
- Verify the component works correctly in both traditional and headless pages.
- Maintain backwards compatibility so the old design continues to work in other parts of the application.

### LINKED ISSUE #30772: UVE: Show "Create New Language Version" Dialog Before Updating Query Params
### Parent Issue

N/A

### Task

Show the "Create New Language Version" dialog when the user clicks on a language where the page does not exist. This will allow us to show the dialog before updating the query parameters, as we do right now, and avoid triggering an unnecessary request if the user chooses not to create the new language version.

### Video

https://github.com/user-attachments/assets/6263d4f5-dda7-4ead-810f-333c1459fdfd

### Proposed Objective

User Experience

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

- Detect when the user clicks on a language where the page does not exist.
- Show the "Create New Language Version" dialog before updating the query parameters.
- Ensure that the dialog allows the user to confirm or cancel the creation of the new language version.
- If the user confirms, proceed with updating the query parameters.
- If the user cancels, do not update the query parameters and avoid triggering the unnecessary request.
- Verify that the application functions correctly after the changes.
- Ensure that all unit tests pass successfully.

### PR DESCRIPTION
**Pull request by KevinDavilaDotCMS**
### Showcase
Traditional: 

https://github.com/user-attachments/assets/4b45d59f-0c5e-4c6c-9242-65223580c0ce

Headless:

https://github.com/user-attachments/assets/cca951fb-d240-4940-af5d-628cde74cf32

This pull request includes significant refactoring and improvements to the `dot-ema-shell.component.ts` and related files. The changes streamline the code, enhance functionality, and improve the testing framework. The most important changes include removing unused services and methods, adding a language selector component, and updating the testing specifications.

### Codebase simplification and cleanup:

* Removed unused imports and services from `dot-ema-shell.component.ts`, including `DotMessageService`, `ConfirmationService`, and `DotLanguage`. [[1]](diffhunk://#diff-677330662fea6dadc7e48fd8455ec2a6fe60d624c7ed1f01f0a3e985aacd05c6L19-L26) [[2]](diffhunk://#diff-677330662fea6dadc7e48fd8455ec2a6fe60d624c7ed1f01f0a3e985aacd05c6L37-R42)
* Deleted the `createNewTranslation` and `shouldNavigate` methods from `dot-ema-shell.component.ts` and replaced them with utility functions `getTargetUrl` and `shouldNavigate`.

### New functionality:

* Added a new `dot-edit-ema-language-selector` component to the toolbar in `dot-uve-toolbar.component.html`. This component allows users to select a language and handles the selection event.
* Implemented the `onLanguageSelected` method in `dot-uve-toolbar.component.ts` to handle language selection and trigger page translation if necessary. [[1]](diffhunk://#diff-217a9e619d6590c4f652e85353b9637ba5e464ddeb0424be35aef39bb8dceb30L2-R23) [[2]](diffhunk://#diff-217a9e619d6590c4f652e85353b9637ba5e464ddeb0424be35aef39bb8dceb30L24-R121)

### Testing improvements:

* Updated the `dot-uve-toolbar.component.spec.ts` to include tests for the new language selector component and its interactions. This includes mocking the `ConfirmationService` and adding tests for language selection scenarios. [[1]](diffhunk://#diff-3eaa147616a5d1ff374a5fa27b0f38f0159a9039ef7e8d672dec43631f48a9e1L1-R10) [[2]](diffhunk://#diff-3eaa147616a5d1ff374a5fa27b0f38f0159a9039ef7e8d672dec43631f48a9e1R39) [[3]](diffhunk://#diff-3eaa147616a5d1ff374a5fa27b0f38f0159a9039ef7e8d672dec43631f48a9e1R55-R57) [[4]](diffhunk://#diff-3eaa147616a5d1ff374a5fa27b0f38f0159a9039ef7e8d672dec43631f48a9e1L129-R138) [[5]](diffhunk://#diff-3eaa147616a5d1ff374a5fa27b0f38f0159a9039ef7e8d672dec43631f48a9e1R179) [[6]](diffhunk://#diff-3eaa147616a5d1ff374a5fa27b0f38f0159a9039ef7e8d672dec43631f48a9e1R204-R225)

These changes collectively improve the maintainability and functionality of the `dot-ema-shell` component and its related parts.

This PR fixes: #30735

### CODE REVIEW

File: core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts
Comment (zJaaal) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts:1]: Awesome!!!

File: core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts
Comment (zJaaal) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts:1]: Nice the shell is less bloated

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts [hunk 1]
```diff
@@ -1393,4 +1413,40 @@ export class EditEmaEditorComponent implements OnInit, OnDestroy {
...
+            rejectIcon: 'hidden',
+            acceptIcon: 'hidden',
+            key: 'shell-confirm-dialog',
+            accept: () => {
+                this.translatePage({ page, newLanguage: language.id });
+            },
+            reject: () => {
+                // If is rejected, bring back the current language on selector
+                this.#goBackToCurrentLanguage();
+                // this.languageSelector().listbox.writeValue(this.$toolbar().currentLanguage);
```
Comment (zJaaal) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts:1435]: remove

File: core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export enum NG_CUSTOM_EVENTS {
     OPEN_WIZARD = 'workflow-wizard',
     DIALOG_CLOSED = 'dialog-closed',
     EDIT_CONTENTLET_UPDATED = 'edit-contentlet-data-updated',
-    URL_IS_CHANGED = 'url-is-changed'
```
Comment (zJaaal) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts:15]: why the change? Should we add a new one instead of modifying this?

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp [hunk 3]
```diff
@@ -724,7 +724,7 @@
                         }
                     };
 
-                    notifyUVEWhenURLisChanged(data);
```
Comment (zJaaal) [inline dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp:727]: same question here

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts (hunk 2 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts:15]: When I added this event/function, I named it that because the ticket was about that topic
But when migrating, I realized that this event refers to create and change the language (not just changing the URL in a new language), and to have more descriptive names, I will change the name. But the behavior is the same

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts (hunk 1 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts]: Done!

File: core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts (hunk 2 above)
Comment (zJaaal) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts:15]: Alrighty, just wanted to be sure nothing broke after this change

**Review by zJaaal — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]
